### PR TITLE
[531] xtable poms are missing license details.

### PR DESCRIPTION
**Pull request by rangareddy**
## *Important Read*

This pull request aims to address issue [#531](https://github.com/apache/incubator-xtable/issues/531)

## What is the purpose of the pull request

Adding the <licenses> tag to xtable project and updating the project name.

## Brief change log

* Added the <licenses> and <organization> in the xtable parent project

```xml
<licenses>
    <license>
        <name>Apache License, Version 2.0</name>
        <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
        <distribution>repo</distribution>
    </license>
</licenses>

<organization>
    <name>The Apache Software Foundation</name>
    <url>https://www.apache.org</url>
</organization>
```

* Updated the project name for each module, for example

```xml
<name>XTable Parent Project</name>
```

## Verify this pull request

Verified by running `$ mvn clean install -DskipTests -U` locally.

### CODE REVIEW

File: xtable-api/pom.xml
Comment (rangareddy) [reply xtable-api/pom.xml:29]: You can refer [Spark](https://github.com/apache/spark/blob/master/streaming/pom.xml#L33) documentation for name tag example.

```xml
  <packaging>jar</packaging>
  <name>Spark Project Streaming</name>
  <url>https://spark.apache.org/</url>
```

File: pom.xml
Comment (rangareddy) [reply pom.xml:30]: It will displays the project name clearly while building the xtable. For Example

```sh
[INFO] XTable Parent Project .............................. SUCCESS [  0.316 s]
[INFO] XTable Api Project ................................. SUCCESS [  0.032 s]
[INFO] XTable Hudi Support Parent Project ................. SUCCESS [  0.025 s]
[INFO] XTable Hudi Support Utils Project .................. SUCCESS [  0.019 s]
[INFO] XTable Core Project ................................ SUCCESS [  0.036 s]
[INFO] XTable Utilities Project ........................... SUCCESS [  0.014 s]
[INFO] XTable Hudi Support Extensions Project ............. SUCCESS [  0.015 s]
[INFO] ------------------------------------------------------------------------
```

File: pom.xml
Comment (rangareddy) [reply pom.xml:22]: Yes it is best practise to keep parent tag at the beginning of the example. Sample example from [Spark](https://github.com/apache/spark/blob/master/streaming/pom.xml#L21)

```
<modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.apache.spark</groupId>
    <artifactId>spark-parent_2.13</artifactId>
    <version>4.0.0-SNAPSHOT</version>
    <relativePath>../pom.xml</relativePath>
  </parent>

  <artifactId>spark-streaming_2.13</artifactId>
```

Comment (vinishjail97) [reply pom.xml:22]: Can you revert your change then ? The below block is at the top now.
```
    <parent>
            <groupId>org.apache</groupId>
        <artifactId>apache</artifactId>
        <version>33</version>
    </parent>
```

Comment (rangareddy) [reply pom.xml:22]: Old [pom.xml](https://github.com/apache/incubator-xtable/blob/main/pom.xml#L30) file

```
    <groupId>org.apache.xtable</groupId>
    <artifactId>xtable</artifactId>
    <name>xtable</name>
    <inceptionYear>2024</inceptionYear>
    <version>0.2.0-SNAPSHOT</version>
    <packaging>pom</packaging>

    <parent>
        <groupId>org.apache</groupId>
        <artifactId>apache</artifactId>
        <version>33</version>
    </parent>
```

New [pom.xml](https://github.com/apache/incubator-xtable/blob/cd395197efdeacd3d01a87c86f74a49ba9409cc3/pom.xml#L22)

```xml
    <parent>
        <groupId>org.apache</groupId>
        <artifactId>apache</artifactId>
        <version>33</version>
    </parent>

    <groupId>org.apache.xtable</groupId>
    <artifactId>xtable</artifactId>
    <name>XTable Parent Project</name>
    <inceptionYear>2024</inceptionYear>
    <version>0.2.0-SNAPSHOT</version>
    <packaging>pom</packaging>
    <url>https://xtable.apache.org/</url>
```

So reverting the code is not required. Please review the pull request once again and approve the pull request.

Comment (vinishjail97) [reply pom.xml:22]: My bad, I was looking at wrong file in spark.

File: xtable-api/pom.xml
Comment (vinishjail97) [reply xtable-api/pom.xml:29]: The new names LGTM, will let @the-other-tim-brown review as well for a second pair of eyes.

Comment (rangareddy) [reply xtable-api/pom.xml:29]: Thanks @vinishjail97 for first level review.

Comment (the-other-tim-brown) [reply xtable-api/pom.xml:29]: Also Api should be API

Comment (rangareddy) [reply xtable-api/pom.xml]: Hi @the-other-tim-brown As for your suggestion, updated the <name> tag like XTable Project [Module] name.

**Review by the-other-tim-brown — APPROVED** [review]

**Comment by rangareddy** [comment]
Hi @vinishjail97 Merge this PR to main branch.

**Pull request merged**
